src/noise.rs

```diff
@@ -0,0 +1,39 @@
+use crate::{FrequencyHz, Signal, TimeSecs};
+
+pub fn sample(rate: FrequencyHz, s: Signal) -> impl Iterator<Item = f64> {
+    (0..).map(move |n: u32| {
+        let sample_period = rate.period();
+        let t = n as f64 * sample_period.0;
+        s.at(TimeSecs(t))
+    })
+}
+
+/// Estimates the root mean square (RMS) of a signal over a period
+///
+/// # Arguments
+///
+/// * `s` - The `Signal` used to compute the RMS
+/// * `period` - The period (starting at phase=0) over which the computation is made
+///
+pub fn rms(s: Signal, period: TimeSecs) -> f64 {
+    // The number of samples should depend on the period.
+    let samples = 100;
+    let rate = FrequencyHz(((samples as f64) / period.0) as u32);
+
+    // The fold function is derived from:
```

Comment:
I really like this comment that gradually morphs the maths into code.
The volume of GitHub-comments are no reflection on the quality of the code comment.